odoo/addons/base/models/ir_qweb.py

```diff
@@ -511,59 +513,60 @@ def keep_query(*keep_params, **additional_params):
                 params[param] = request.httprequest.args.getlist(param)
     return werkzeug.urls.url_encode(params)
 
+
 ####################################
-###        QWebException         ###
+###             QWeb             ###
 ####################################
 
-class QWebException(Exception):
-    """ Management of errors that raised when rendering a QWeb template.
-    """
-    def __init__(self, message, qweb, template=None, ref=None, path_xml=None, code=None):
-        self.stack = traceback.format_exc()
-        self.name = template
-        self.ref = ref
-        self.path, self.html = path_xml or (None, None)
-        self.code = None
-        if code:
-            self.code = '\n'.join(code.split('\n')[:-1]) if qweb.env.context.get('dev_mode') else None
-            line_nb = 0
-            for error_line in reversed(self.stack.split('\n')):
-                if f'File "<{self.ref}>"' in error_line:
-                    line_function = error_line.split(', line ')[1]
-                    line_nb = int(line_function.split(',')[0])
-                    break
-            for code_line in reversed(code.split('\n')[:line_nb]):
-                match = re.match(r'\s*# element: (.*) , (.*)', code_line)
-                if match:
-                    self.path = match[1][1:-1]
-                    self.html = match[2][1:-1]
-                    break
 
-        self.title = message
-        super().__init__(message)
+class QWebError(Exception):
+    qweb: QWebErrorInfo | None
```

Comment:
You can remove it. `qweb` is set on all exceptions and this one does not always initialize it.
```suggestion
```